src/c/core/session/session.c

```diff
@@ -748,7 +748,15 @@ bool wait_session_status(
     for (size_t i = 0; i < attempts && session->info.last_requested_status == UXR_STATUS_NONE; ++i)
     {
         send_message(session, buffer, length);
-        listen_message(session, UXR_CONFIG_MIN_SESSION_CONNECTION_INTERVAL);
+
+        int64_t start_timestamp = uxr_millis();
+        int remaining_time = UXR_CONFIG_MIN_SESSION_CONNECTION_INTERVAL;
+
+        do
+        {
+            listen_message(session, remaining_time);
+            remaining_time = UXR_CONFIG_MIN_SESSION_CONNECTION_INTERVAL - (int)(uxr_millis() - start_timestamp);
```

Comment:
This causes invalid underflow when the `uxr_millis()` returns something invalid, such as the embedded device does not have GPS time yet. I could use some info on how this is being used internally if we have to have a time jump (from boot time to RTC time)

Reply:
Which implementation of uxr_millis() are you using?

Reply:
Our software runs on multiple OS's. This one is ChibiOS, so we had to write our own implentation of `uxr_millis()`. It was not clear from the public API on the requirements for that function. I filed this bug for more info:
https://github.com/eProsima/Micro-XRCE-DDS-Client/issues/350